src/Data/Array.purs

```diff
@@ -229,7 +229,8 @@ foreign import length :: forall a. Array a -> Int
 -- | ```
 -- |
 -- | Note, the running time of this function is `O(n)`.
-foreign import cons :: forall a. a -> Array a -> Array a
+cons :: forall a. a -> Array a -> Array a
+cons x xs = ST.run (STA.withArray (STA.splice 0 0 [x]) xs)
```

Comment:
I think `[x] <> xs` would be equally good as the implementation here, and same for snoc.

Reply:
I think that's a good idea for `cons`, but I tried to benchmark `.concat` which `<>` uses and `.slice().push()` that we use for `snoc` and `.slice().push()` seems faster around 75% in chrome